### ISSUE TITLE
[WriteSupport] Write a dataframe to Neo4j via provided query

### ISSUE THREAD

**Issue by conker84** [neo4j-contrib/neo4j-spark-connector#143]
We want to ingest the a dataframe into Neo4j via provided query in the following way:
```scala
otherFormatDf
   .write
   .format("neo4j")
   .option("query", "MERGE (p:Person {ssn: row.ssn}) SET p.name = row.fullname")
   .save()
```
where `row` is the Dataframe row value provided as map.

Under-the-hood we'll use the same approach as the Neo4j Streams plugin so we'll append the provided query to an UNWIND operation:

```cypher
UNWIND batch AS row
MERGE (p:Person {ssn: row.ssn}) SET p.name = row.fullname
```

The default behaviour will use the Dataframe partition size as batch size, otherwise the user can provide an option param where sets the batch size.

**Comment by moxious** [comment]
Can we please bind it to `event` and not to `row`?  I I know that "row" feels more natural for spark, but it would be very nice for it to work the same way as the cyphersink does from the kafka plugin.

**Comment by moxious** [comment]
Also see comments on other tickets.  There needs to be an explicit TX under the covers with the neo4j driver, and there is an implied option of write that needs to get passed to the driver to make this work.

**Comment by conker84** [comment]
Two approaches:
1. Provide only strings
2. Ask to the user to provide the schema

**Comment by conker84** [comment]
closed via 609cd0e
